### PR TITLE
add check instances in Service in autoscalinggroup

### PR DESCRIPTION
**Pull request by phoppe93**
#### General
- [ ] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)
- [ ] Update README with any necessary configuration snippets
- [ ] Binstubs are created if needed
- [ ] RuboCop passes
- [ ] Existing tests pass 
#### New Plugins
- [ ] Tests
- [ ] Add the plugin to the README
- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

File: bin/check-asg-instances-inservice.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+#   Peter Hoppe <peter.hoppe.extern@bertelsmann.de>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'aws-sdk'
+
+class CheckAsgInstancesInService < Sensu::Plugin::Check::CLI
+  option :aws_access_key,
```
Comment (sstarcher) [inline bin/check-asg-instances-inservice.rb]: remove these as the aws tool supports them directly from the environment.

File: bin/check-asg-instances-inservice.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+         description: 'AWS Region (defaults to us-east-1).',
+         default:     'us-east-1'
+
+  option :group,
+         short:       '-g G',
+         long:        '--autoscaling-group GROUP',
+         description: 'AutoScaling group to check'
+
+
+  def aws_config
```
Comment (sstarcher) [inline bin/check-asg-instances-inservice.rb]: not needed

File: bin/check-asg-instances-inservice.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    { access_key_id: config[:aws_access_key],
+      secret_access_key: config[:aws_secret_access_key],
+      region: config[:aws_region] }
+  end
+
+  def asg
+    @asg ||= Aws::AutoScaling::Client.new aws_config
```
Comment (sstarcher) [inline bin/check-asg-instances-inservice.rb]: you can just create the client no need to call aws_config

File: bin/metrics-rds.rb [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+require 'sensu-plugins-aws'
+require 'sensu-plugin/metric/cli'
+require 'aws-sdk'
+require 'time'
+
+class RDSMetrics < Sensu::Plugin::Metric::CLI::Graphite
+  include Common
+
+  option :aws_access_key,
```
Comment (sstarcher) [inline bin/metrics-rds.rb]: same as above

File: bin/check-asg-instances-inservice.rb (hunk 2 above)
Comment (runningman84) [inline bin/check-asg-instances-inservice.rb]: Checking specific groups is a use case here.

**Comment by phoppe93** [comment]
@sstarcher  

Hey, i used you comments to fix the code. 

And i add a new check for the loadBalancer, please check it.

File: bin/check-asg-instances-inservice.rb [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+#   Peter Hoppe <peter.hoppe.extern@bertelsmann.de>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'aws-sdk'
+
+class CheckAsgInstancesInService < Sensu::Plugin::Check::CLI
+  option :aws_region,
```
Comment (sstarcher) [inline bin/check-asg-instances-inservice.rb]: Include Common similar to https://github.com/sensu-plugins/sensu-plugins-aws/blob/master/bin/check-eip-allocation.rb#L34

File: bin/check-asg-instances-inservice.rb [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+         description: 'AWS Region (defaults to us-east-1).',
+         default:     ENV['AWS_REGION']
+
+  option :group,
+         short:       '-g G',
+         long:        '--autoscaling-group GROUP',
+         description: 'AutoScaling group to check'
+
+  def asg
+    Aws.config[:region] = config[:aws_region]
```
Comment (sstarcher) [inline bin/check-asg-instances-inservice.rb]: this line can be removed as common will handle this

File: bin/check-asg-instances-inservice.rb [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    @asg ||= Aws::AutoScaling::Client.new(region: config[:aws_region])
```
Comment (sstarcher) [inline bin/check-asg-instances-inservice.rb]: remove the arguments

File: bin/check-elb-instances-inservice.rb [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+#   Peter Hoppe <peter.hoppe.extern@bertelsmann.de>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'aws-sdk'
+
+class CheckElbInstanceInService < Sensu::Plugin::Check::CLI
+  option :aws_region,
```
Comment (sstarcher) [inline bin/check-elb-instances-inservice.rb]: same as above

File: bin/check-elb-instances-inservice.rb [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+         description: 'AWS Region (defaults to us-east-1).',
+         default:     ENV['AWS_REGION']
+
+  option :load_balancer ,
+         short:       '-l L',
+         long:        '--load_balancer LoudBalancer',
+         description: 'LoadBalancer Name to check'
+
+  def elb
+    Aws.config[:region] = config[:aws_region]
```
Comment (sstarcher) [inline bin/check-elb-instances-inservice.rb]: same

File: bin/check-elb-instances-inservice.rb [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+    @elb ||= Aws::ElasticLoadBalancing::Client.new(region: config[:aws_region])
```
Comment (sstarcher) [inline bin/check-elb-instances-inservice.rb]: same

File: bin/metrics-rds.rb [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+#
+
+require 'sensu-plugins-aws'
+require 'sensu-plugin/metric/cli'
+require 'aws-sdk'
+require 'time'
+
+class RDSMetrics < Sensu::Plugin::Metric::CLI::Graphite
+
+  option :aws_access_key,
```
Comment (sstarcher) [inline bin/metrics-rds.rb]: drop this option

File: bin/metrics-rds.rb [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+         short:       '-a AWS_ACCESS_KEY',
+         long:        '--aws-access-key AWS_ACCESS_KEY',
+         description: "AWS Access Key. Either set ENV['AWS_ACCESS_KEY'] or provide it as an option",
+         default:     ENV['AWS_ACCESS_KEY']
+
+  option :aws_secret_access_key,
```
Comment (sstarcher) [inline bin/metrics-rds.rb]: drop this option

File: bin/metrics-rds.rb [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+
+  option :statistics,
+         short:       '-S N',
+         long:        '--statistics NAME',
+         default:     :average,
+         proc:        proc { |a| a.downcase.intern },
+         description: 'CloudWatch statistics method'
+
+  def rds
+    @rds = Aws::RDS::Client.new(aws_config)
```
Comment (sstarcher) [inline bin/metrics-rds.rb]: drop the args

**Comment by phoppe93** [comment]
Hey @sstarcher  please check it again.

I add another new check, to check how many instances at a autoscalinggroup are terminated and launched.

Thank you for check

**Comment by sstarcher** [comment]
Can you rebase and a few of my comments still need addressed.

File: bin/check-asg-instances-inservice.rb (hunk 6 above)
Comment (phoppe93) [inline bin/check-asg-instances-inservice.rb]: removed aws config at
bin/check-asg-instances-created.rb
bin/check-asg-instances-inservice.rb
bin/metrics-rds.rb
bin/check-elb-instances-inservice.rb

File: bin/check-asg-instances-inservice.rb (hunk 7 above)
Comment (phoppe93) [inline bin/check-asg-instances-inservice.rb]: arguments removed

File: bin/metrics-rds.rb [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+  end
+
+  def cloud_watch
+    @cloud_watch = Aws::CloudWatch::Client.new(aws_config)
```
Comment (phoppe93) [inline bin/metrics-rds.rb]: drop the args aws_config

File: bin/metrics-rds.rb [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugins-aws'
+require 'sensu-plugin/metric/cli'
+require 'aws-sdk'
+require 'time'
+
+class RDSMetrics < Sensu::Plugin::Metric::CLI::Graphite
```
Comment (phoppe93) [inline bin/metrics-rds.rb]: Common added at 

bin/check-asg-instances-created.rb
bin/check-asg-instances-inservice.rb
bin/metrics-rds.rb
bin/check-elb-instances-inservice.rb

File: bin/metrics-rds.rb (hunk 11 above)
Comment (phoppe93) [inline bin/metrics-rds.rb]: aws_access_key 
aws_secret_access_key

deleted at 
bin/check-asg-instances-created.rb
bin/check-asg-instances-inservice.rb
bin/metrics-rds.rb
bin/check-elb-instances-inservice.rb

**Comment by phoppe93** [comment]
@sstarcher please check the comments

**Comment by sstarcher** [comment]
@phoppe93 looks good.  Could you rebase off master to resolve the merge conflict.

**Comment by phoppe93** [comment]
@sstarcher merge conficts are resolved

**Pull request merged**